### PR TITLE
[FIRRTL] Type lowering of operations with no name

### PR DESCRIPTION
**Pull request by prithayan**
Check if name attribute exists, before accessing it while lowering register and registerReset op.
If no name exists in aggregate operation, then no name should be added after lowering. 
Fix for: https://github.com/llvm/circt/issues/795. 

With name attribute, 
``` mlir 
%r = firrtl.regreset %clock, %reset, %init {name = "r"} : (!firrtl.clock, !firrtl.uint<1>, !firrtl.vector<uint<1>, 2>) -> !firrtl.vector<uint<1>, 2>
``` 
is lowered to: 
   ``` mlir
%r_0 = firrtl.regreset %clock, %reset, %init_0 {name = "r_0"} : (!firrtl.clock, !firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
%r_1 = firrtl.regreset %clock, %reset, %init_1 {name = "r_1"} : (!firrtl.clock, !firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
```

Without name attribute 
``` mlir    
 %r = firrtl.reg %clock : (!firrtl.clock) -> !firrtl.vector<uint<1>, 2> 
``` 
is lowered to 
``` mlir
%0 = firrtl.reg %clock : (!firrtl.clock) -> !firrtl.uint<1>
%1 = firrtl.reg %clock : (!firrtl.clock) -> !firrtl.uint<1>
```

### CODE REVIEW

File: lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp [hunk 1]
```diff
@@ -571,12 +571,17 @@ void TypeLoweringVisitor::visitDecl(RegOp op) {
 
   // Loop over the leaf aggregates.
   for (auto field : fieldTypes) {
-    SmallString<16> loweredName(op.nameAttr().getValue());
-    loweredName += field.suffix;
+    SmallString<16> loweredName("");
```
Comment (lattner) [inline lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp:574]: How about structuring this as:

```
 StringAttr loweredName;
 if (auto nameAttr = op.nameAttr())
      loweredName = builder->getStringAttr(nameAttr.getValue().str+field.suffix);
  ```

that will eliminate the extra check in the create call.

**Review by lattner — APPROVED** [review]
Looks good with the simplification I suggested (in both places).  Thanks for fixing this!

File: lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp (hunk 1 above)
Comment (prithayan) [reply lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp]: Thanks, simplified it.

**Review by lattner — APPROVED** [review]
looks great!

**Pull request merged**